python/ptest/cases/dcdc_checkout_case.py

```diff
@@ -0,0 +1,114 @@
+
+    @disable_cmd.setter 
+    def disable_cmd(self, value): 
+        assert(value == "true" or value == "false")
+        self.simulation.flight_controller.write_state("dcdc.disable_cmd", value)
+
+    @reset_cmd.setter 
+    def reset_cmd(self, value): 
+        assert(value == "true" or value == "false")
+        self.simulation.flight_controller.write_state("dcdc.reset_cmd", value)
+
+    def setup_case_singlesat(self, simulation):
+        self.simulation=simulation
+        self.simulation.flight_controller.write_state(
+            "pan.state", 11)  # Manual state
+        self.run_case_singlesat()
+        print("DCDC cases finished.")
+
+    def run_case_singlesat(self):
+        self.simulation.cycle_no = self.simulation.flight_controller.read_state("pan.cycle_no")
+
+        if self.adcs_cmd==self.adcs_rd and self.sph_cmd==self.sph_rd and self.disable_cmd=="false" and self.reset_cmd=="false":
+            print("Control task initialized correctly") 
+
+        print ("Test case 1: Try both DCDCs on. Turn on all systems (ADCS motors, prop valves, docking motor).")
+
+        self.adcs_cmd="true"
+        self.sph_cmd="true"
+
+        if self.adcs_rd=="true" and self.sph_rd=="true":
+            print("Passed")
+        else:
+            print("Failed")
+
+        print("Test case 2: ADCS DCDC on, SPH + Prop DCDC off. Turn on all systems.")
+
+        self.adcs_cmd="true"
+        self.sph_cmd="false"
+        if self.adcs_rd=="true" and self.sph_rd=="false":
+            self.reset_cmd="true"
+            # Reset takes at least one control cycle to complete
+            self.simulation.flight_controller.write_state("cycle.start", "true")
+            if (self.adcs_rd == "true" and self.sph_rd == "true"):
+                print("Passed")
+            else:
+                print("Unable to reset pins")
+        else:
+            print("Unable to turn ADCS Motor on and SpikeDock off")
+
+        print("Test Case 3: ADCS DCDC off, SPH + Prop DCDC on. Turn on all systems. ")
+
+        self.adcs_cmd="false"
+        self.sph_cmd="true"
+        if self.adcs_rd=="false" and self.sph_rd=="true":
+            self.reset_cmd="true"
+            # Reset takes at least one control cycle to complete
+            self.simulation.flight_controller.write_state("cycle.start", "true")
+            if (self.adcs_rd == "true" and self.sph_rd == "true"):
+                print("Passed")
+            else:
+                print("Unable to reset pins")
+        else:
+            print("Unable to turn ADCS Motor off and SpikeDock on")
+
+        print("Test Case 4: ADCS DCDC off, SPH + Prop DCDC off. Turn on all systems. ")
+
+        self.disable_cmd="true"
+        if self.adcs_rd=="false" and self.sph_rd=="false":
+            self.reset_cmd="true"
+            # Reset takes at least one control cycle to complete
+            self.simulation.flight_controller.write_state("cycle.start", "true")
```

Comment:
You can probably make this a property or function in the base-level case file (base.py)